File: src/main/java/by/sorface/gateway/config/properties/SessionProperties.java
package by.sorface.gateway.config.properties;

import org.springframework.boot.context.properties.ConfigurationProperties;
import java.time.Duration;

@ConfigurationProperties("spring.session")
public class SessionProperties {
    private CookieProperties cookie = new CookieProperties();

    public CookieProperties getCookie() {
        return cookie;
    }

    public void setCookie(CookieProperties cookie) {
        this.cookie = cookie;
    }

    public static class CookieProperties {
        private String name = "gtw_sid";
        private String domain = "localhost";
        private String path = "/";
        private boolean httpOnly = true;
        private boolean secure = false;
        private String sameSite = "Lax";
        private Duration maxAge = Duration.ofSeconds(1000000000);

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }

        public String getDomain() {
            return domain;
        }

        public void setDomain(String domain) {
            this.domain = domain;
        }

        public String getPath() {
            return path;
        }

        public void setPath(String path) {
            this.path = path;
        }

        public boolean isHttpOnly() {
            return httpOnly;
        }

        public void setHttpOnly(boolean httpOnly) {
            this.httpOnly = httpOnly;
        }

        public boolean isSecure() {
            return secure;
        }

        public void setSecure(boolean secure) {
            this.secure = secure;
        }

        public String getSameSite() {
            return sameSite;
        }

        public void setSameSite(String sameSite) {
            this.sameSite = sameSite;
        }

        public Duration getMaxAge() {
            return maxAge;
        }

        public void setMaxAge(Duration maxAge) {
            this.maxAge = maxAge;
        }
    }
} 

File: src/main/kotlin/by/sorface/gateway/config/properties/SessionProperties.kt
package by.sorface.gateway.config.properties

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.bind.ConstructorBinding
import java.time.Duration

/**
 * Configuration properties for session management.
 */
@ConfigurationProperties(prefix = "spring.session")
data class SessionProperties(
    /**
     * Cookie configuration properties for the session.
     */
    val cookie: CookieProperties = CookieProperties()
) {
    /**
     * Configuration properties for session cookies.
     */
    data class CookieProperties(
        /**
         * The name of the session cookie.
         * Default value is "gtw_sid".
         */
        val name: String = "gtw_sid",
        
        /**
         * The domain of the session cookie.
         * Default value is "localhost".
         */
        val domain: String = "localhost",
        
        /**
         * The path of the session cookie.
         * Default value is "/".
         */
        val path: String = "/",
        
        /**
         * Whether the session cookie should be HTTP only.
         * Default value is true.
         */
        val httpOnly: Boolean = true,
        
        /**
         * Whether the session cookie should be secure.
         * Default value is false.
         */
        val secure: Boolean = false,
        
        /**
         * The SameSite attribute of the session cookie.
         * Default value is "Lax".
         */
        val sameSite: String = "Lax",
        
        /**
         * The maximum age of the session cookie in seconds.
         * Default value is 1000000000 seconds.
         */
        val maxAge: Duration = Duration.ofSeconds(1000000000)
    )
} 